### ISSUE TITLE
lia needs remember

### ISSUE THREAD

**Issue by roglo** [coq/coq#11270]
#### Description of the problem

```coq
Require Import Psatz.
Theorem foo : forall a b, 1 <= S (a + a * S b).
Proof.
intros.
lia.
```

> Error: Tactic failure:  Cannot find witness.

```coq
remember (a * S b).
lia.
```

> No more subgoals.

#### Coq Version

> The Coq Proof Assistant, version 8.12+alpha (December 2019)
> compiled on Dec 10 2019 9:08:09 with OCaml 4.07.2+dev0-2018-10-04

**Comment by olaure01** [comment]
It seems to be a regression: it works with Coq 8.10.2 but not with 8.11+beta1.

**Comment by Zimmi48** [comment]
cc @fajb: do you have any idea what might have caused this?

**Comment by fajb** [comment]
This is a bad interaction between the new zify and infamous PR #9725.
This is fixed for lia by PR #11263 which reverts the problematic part of  PR #9725.

Yet, this also impacts omega. After the new zify, we get the following goal
```  a, b : nat
  cstr, cstr0 : (0 <= Z.of_nat a)%Z
  cstr1 : (0 <= Z.of_nat b)%Z
  ============================
  (1 <= Z.of_nat a + Z.of_nat a * (Z.of_nat b + 1) + 1)%Z
```
which needs some (limited) non-linear reasoning namely Z.of_nat a * Z.of_nat b >= 0 which is internally performed by lia. This used to be generated by zify in 8.10 i.e.,
`(0 <= Z.of_nat a * Z.succ (Z.of_nat b))%Z `
